hail/python/hail/expr/functions.py

```diff
@@ -2971,6 +2971,33 @@ def len(x) -> Int32Expression:
         return apply_expr(lambda x: ArrayLen(x), tint32, array(x))
 
 
+@typecheck(x=expr_oneof(expr_array(), expr_str))
+def reversed(x):
+    """Reverses the elements of a collection.
+
+    Examples
+    --------
+    >>> a = ['The', 'quick', 'brown', 'fox']
+    >>> hl.eval(hl.reversed(a))
+    ['fox', 'brown', 'quick', 'The']
+
+    Parameters
+    ----------
+    x : :class:`.ArrayExpression` or :class:`.StringExpression`
+        Array or string expression.
+
+    Returns
+    -------
+    :class:`.Expression`
+    """
+
+    typ = x.dtype
+    x = range(0, len(x)).map(lambda i: x[len(x) - 1 - i])
+    if typ == tstr:
```

Comment:
`typ` is only used once, so just inline `x.dtype` here.

Reply:
I can't inline the type because x is mutated and will always be Array[String] after line 2 even if the input is a string.